README.md

```diff
@@ -1,35 +1,251 @@
+    ```
+
+## Tool Documentation
+At Codacy we strive to provide the best value to our users and, to accomplish that, we document our patterns so that the user can better understand the problem and fix it.
+
+At this point, your tool has everything it needs to run, but there is one other really important thing that you should do before submitting your docker: the documentation for your tool.
 
-[Scala Docker Template Docs](http://docs.codacy.com/v1.5/docs/tool-developer-guide-using-scala)
+Your files for this section should be placed in /docs/description/.
+
+In order to provide more details you can create:
+* A single /docs/description/description.json
+* **(Optional)** A /docs/description/<PATTERN-ID>.md for each pattern. 
+This documentation should also be generated automatically to avoid having to go through all of the files each time it needs to be updated.
+
+In the description.json you define the title for the pattern, brief description, time to fix (in minutes), and also a description of the parameters in the following format:    
+
+
+```json
+[
+  {
+    "patternId":"latedef",
+    "title":"This is a title",
+    "description":"This is a description",
+    "timeToFix": 5,
+    "parameters":[
+      {
+        "name":"latedef",
+        "description":"this is a param description"
+      }
+    ]
+  }
+]
+```
+
+**(Optional)** To give a more detailed explanation about the issue, you should define the <PATTERN-ID>.md
+
+```markdown
+Fields in interfaces are automatically public static final, and methods are public abstract.
+Classes or interfaces nested in an interface are automatically public and static (all nested interfaces are automatically static).
+
+For historical reasons, modifiers which are implied by the context are accepted by the compiler, but are superfluous.
+
+Ex:
+
+    public interface Foo {
+        public abstract void bar();         // both abstract and public are ignored by the compiler
+        public static final int X = 0;         // public, static, and final all ignored
+        public static class Bar {}             // public, static ignored
+        public static interface Baz {}         // ditto
+        
+        void foo();                            //this is correct
+    }
+
+    public class Bar {
+        public static interface Baz {} // static ignored
+    }
+
+[Source](http://pmd.sourceforge.net/pmd-5.3.2/pmd-java/rules/java/unusedcode.html#UnusedModifier)
+```
+
+You should explain the what and why of the issue. Adding an example is always a nice way to help other people understand the problem. For a more thorough explanation you can also add a link at the end referring a more complete source.
 
 ## Test
 
-Follow the instructions at [codacy-plugins-test](https://github.com/codacy/codacy-plugins-test/blob/master/README.md#test-definition)
+Follow the instructions at [codacy-plugins-test](https://github.com/codacy/codacy-plugins-test/blob/master/README.md#test-definition).
+
+## Submit the docker
```

Comment:
Something about publishing the docker using the seed is missing

Reply:
I added on the setup section a reference to the sbt-native-packager and to the docker:publishLocal command.